test/common/testlib.py

```diff
@@ -854,7 +854,9 @@ def label(self):
     def new_machine(self, image=None, forward={}, restrict=True, cleanup=True, **kwargs):
         machine_class = self.machine_class
         if image is None:
-            image = self.image
+            image = os.path.join(TEST_DIR, "images", self.image)
+            if not os.path.exists(image):
+                raise FileNotFoundError("Can't run tests without a prepared image; use test/image-prepare")
```

Comment:
Move this after the "else:" of `if opts.address`, then it should be good.

Reply:
Thanks.  I added back the `image or self.image` logic for the `--machine` case.